app/src/main/java/org/mozilla/tv/firefox/IntentValidator.kt

```diff
@@ -57,10 +60,18 @@ object IntentValidator {
 
         when (intent.action) {
             Intent.ACTION_MAIN -> {
-                val dialParams = intent.extras?.getString(DIAL_PARAMS_KEY) ?: return null
-                if (dialParams.isNotEmpty()) {
+                val extras = intent.extras ?: return null
+
+                if (extras.keySet().contains(DIAL_PARAMS_KEY)) {
+                    val param = extras.getString(DIAL_PARAMS_KEY)
                     TelemetryIntegration.INSTANCE.youtubeCastEvent()
-                    return ValidatedIntentData(url = "https://www.youtube.com/tv?$dialParams", source = Session.Source.ACTION_VIEW)
+                    return ValidatedIntentData(url = "https://www.youtube.com/tv?$param", source = Session.Source.ACTION_VIEW)
+                }
+                // The fetch delay can be changed for testing purposes through an intent with custom flags
+                // isQA and fetchDelay. Where isQA is true/false and fetchDelay is the intended next fetch time
+                // in seconds.
+                else if (extras.keySet().contains(IS_QA_BUILD_KEY)) {
+                    PocketVideoFetchScheduler.setDelayForQA(extras.getLong(FETCH_DELAY_KEY))
```

Comment:
Static state is really hard to reason about and should be avoided! For example, if our process does not die but the application is recreated (in practice, not sure if this ever happens), the static state will remain even though we have a new scheduler: ouch!

I think we can replace this with `context.serviceLocator.pocketFetchScheduler...`